packages/vite/src/node/plugins/json.ts

```diff
@@ -47,22 +48,46 @@ export function jsonPlugin(
       json = stripBomTag(json)
 
       try {
-        if (options.stringify) {
-          if (isBuild) {
-            return {
-              // during build, parse then double-stringify to remove all
+        const stringify =
+          options.stringify === true ||
+          // use 10kB as a threshold
+          // https://v8.dev/blog/cost-of-javascript-2019#:~:text=A%20good%20rule%20of%20thumb%20is%20to%20apply%20this%20technique%20for%20objects%20of%2010%20kB%20or%20larger
+          (options.stringify === 'auto' && json.length >= 10 * 1000)
```

Comment:
Maybe we should benchmark this to find out whether the threshold has changed over the last five years.

Reply:
I ran the benchmark: https://github.com/sapphi-red/json-parse-benchmark

It's difficult to measure where the threshold is due to the overhead of starting the JS engines. But the perf benefit still exists. Probably fine to go with 10kB for now.